corehq/apps/userreports/specs.py

```diff
@@ -13,11 +19,57 @@ def TypeProperty(value):
     return StringProperty(required=True, choices=[value])
 
 
-class FactoryContext(namedtuple('FactoryContext', ('named_expressions', 'named_filters'))):
+@dataclass
+class FactoryContext():
+    named_expressions: dict
+    _named_expressions: dict = field(init=False, repr=False)
+
+    named_filters: dict
+    _named_filters: dict = field(init=False, repr=False)
+
+    domain: Optional[str] = None
+
+    @property
+    @memoized
+    def named_filters(self):
```

Comment:
How do these functions interact with the attributes created by dataclasses? With these additions I wonder if a vanilla class wouldn't be easier to understand?

Reply:
Yeah, it still works with the dataclass attribute (the constructor still works as it would with a regular dataclass). Would be happy to convert it to a regular class if you think worthwhile